_posts/2021-04-30-lessons-learned-online-experiments.md

```diff
@@ -0,0 +1,193 @@
+
+<p align="center">
+    <img width="60%" src="{{ site.baseurl }}{% link images/lessons-learned-online-experiments/skewed_p_values.png %}">
+</p>
+
+The intuition behind this is that your results could be highly influenced by which users land in which group. If you have some users with a lot of sessions, and a really high or low session conversion rate, that could heavily influence the overall session conversion rate for that group.
+
+Rather than throwing my hands up and changing the strategy, I decided to run a simulation to see if we would actually be impacted by this. The idea behind the simulation was to take our population and simulate many experiments where we randomized by user and compared session conversion rates like we were planning to do in our real experiment. I then checked if we actually saw a higher rate of false positives, and it turned out we did not, so we decided to stick with our existing plan.<sup>3</sup>
+
+The key lesson here was that **💡simulations are your friend**. If you're ever unsure about some statistical effect, it's very quick (and fun) to run a simulation to see how you'd be affected, before jumping to any conclusions.
+
+# Influencing big & small decisions
+
+Data is commonly used to influence big decisions with an obvious need for quantitative evidence. Does this feature positively impact our users? Should we roll it out to everyone? But there is also a large realm of much smaller decisions that could be equally influenced by data.
+
+Around the time we were planning one of our experiments, the system responsible for rendering the relevant website content was in the process of being upgraded. The legacy system ("Renderer 1") was still handling ~15% of the traffic, while the new system ("Renderer 2") was handling the other 85%.
+
+This posed a question to us: do we need to implement our experiment in the two different codebases for each rendering system? Based on the sizeable 15% still going to "Renderer 1", our initial thinking was yes. However, we decided to dig a bit deeper.
+
+<p align="center">
+    <img width="75%" src="{{ site.baseurl }}{% link images/lessons-learned-online-experiments/renderer_1.png %}">
+</p>
+
+With our experiment design, we'd only be giving the buyers the treatment/control experience on the first request in a given session. With that in mind, the question we actually need to answer changed. Instead of what of % all requests across all users are served by "Renderer 2", we needed to look what % of first requests in a session are served by "Renderer 2", for the users we planned to include in our experiment.
```

Comment:
```suggestion
With our experiment design, we'd only be giving the users the treatment/control experience on the first request in a given session. With that in mind, the question we actually need to answer changed. Instead of what of % all requests across all users are served by "Renderer 2", we needed to look what % of first requests in a session are served by "Renderer 2", for the users we planned to include in our experiment.
```